pkg/movingaverage/median_filter.go

```diff
@@ -24,6 +26,9 @@ type MedianFilter struct {
 	size          uint64
 	count         uint64
 	instantaneous float64
+	g             uint64
```

Comment:
Now the computation becomes so complex that a lock is required

Reply:
If we need to consider data race, locks are needed regardless of the implementation of the median algorithm, so we can consider locks in another PR. Here we only need to consider  not causing panic or extreme wrong data